rfd/0003-teleport-plugins-versioning.md

```diff
@@ -0,0 +1,83 @@
+---
+authors: Brian Joerger (bjoerger@goteleport.com)
+state: draft
+---
+
+# RFD 3 - Teleport Plugins releases
+
+## What
+
+Release strategy for Teleport plugins.
+
+## Why
+
+With the release of the Teleport API as a [go module](https://pkg.go.dev/github.com/gravitational/teleport)
+and the addition of new plugins (Terraform) which must be updated for every new 
+version, it's time to layout a proper release process.
+
+## Details
+
+Teleport plugins are released in line with Teleport releases. This leads to the
+plugins following the same [versioning scheme](https://github.com/gravitational/teleport/blob/master/rfd/0012-teleport-versioning.md#rfd-12---teleport-versioning)
+as Teleport, and more importantly, the same [compatibility guidelines](https://github.com/gravitational/teleport/blob/master/rfd/0012-teleport-versioning.md#compatibility).
+
+These releases are more often than not just vanity releases, with no functional
+changes in them. When there are functional changes in a plugin, they must follow 
+the same compatibility guidelines above.
+
+### Release process
+
+The full release process is contained in `make release-version VERSION=vX.Y.Z`.
+This will update the version files of each plugin, update the `github/gravitational/teleport/api`
+dependency, regenerated the Terraform Schema, and update the git tags.
+
+Some basic manual changes may need to be made to handle significant changes
+made to the Teleport API, as explained below.
```

Comment:
We also need to update `TELEPORT_GET_VERSION` in integration tests, as well as `lib/testing/integration/download.downloadVersions`, at least for major versions.